### PR TITLE
[Tests-Only] Adjust user roles in API acceptance tests

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
Adjusts acceptance test scenarios so that the "standard" usernames are used in a standard way:

- tests that have just a single ordinary user use `user0`
- sharing tests have `user0` share to `user1`
- resharing tests then have `user` resharing to `user2` ...
- tests that create and test an extra admin account use `another-admin` or a subadmin use `subadmin` `another-subadmin`
- tests that create and modify attributes of an ordinary account use `brand-new-user` then `another-new-user` and similar for groups `brand-new-group` and `another-new-group` (e.g. Provisioning API tests) (names subject to change!)
- tests that check API calls to a non-existent user use `nonexistentuser` and for groups `nonexistentgroup`

The "primary actor" in most scenarios is either `admin` or `user0`

This will make it easier to choose substitute usernames (UIDs) and be confident that all the test scenarios are being run in some sensible way with the substitutions.

This PR has changes for test suites apiAuth through to apiShareManagementBasic.

## Related Issue
#33596 

## How Has This Been Tested?
CI

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Review by individual-it — APPROVED** [review]
looks good, but please make a PR to ocis & ocis-reva with this branch as test-runner to prove that it works there

File: tests/acceptance/features/apiProvisioning-v1/getUser.feature [hunk 1]
```diff
@@ -53,56 +53,56 @@ Feature: get user
...
     Then the OCS status code should be "100"
     And the HTTP status code should be "200"
     And the display name returned by the API should be "New User"
     And the quota definition returned by the API should be "default"
 
   Scenario: a normal user gets their own information, providing uppercase username as authentication
     Given these users have been created with default attributes and skeleton files:
-      | username | displayname |
-      | newuser  | New User    |
-    When user "NEWUSER" retrieves the information of user "newuser" using the provisioning API
```
Comment (phil-davis) [inline tests/acceptance/features/apiProvisioning-v1/getUser.feature:105]: The uppercase has got lost here - see issue #38014

File: tests/acceptance/features/apiProvisioning-v1/getUser.feature [hunk 2]
```diff
@@ -111,9 +111,9 @@ Feature: get user
   @skipOnOcV10.3
   Scenario: a normal user gets their own information, providing uppercase username in the URL
     Given these users have been created with default attributes and skeleton files:
-      | username | displayname |
-      | newuser  | New User    |
-    When user "newuser" retrieves the information of user "NEWUSER" using the provisioning API
```
Comment (phil-davis) [inline tests/acceptance/features/apiProvisioning-v1/getUser.feature:116]: The uppercase has got lost here - see issue #38014

File: tests/acceptance/features/apiProvisioning-v1/getUser.feature [hunk 3]
```diff
@@ -122,19 +122,19 @@ Feature: get user
   @skipOnOcV10.3
   Scenario: a mixed-case normal user gets their own information, providing lowercase username in the URL
     Given these users have been created with default attributes and skeleton files:
-      | username | displayname |
-      | NewUser  | New User    |
-    When user "NewUser" retrieves the information of user "newuser" using the provisioning API
```
Comment (phil-davis) [inline tests/acceptance/features/apiProvisioning-v1/getUser.feature:127]: The mixed  case has got lost here - see issue #38014

File: tests/acceptance/features/apiProvisioning-v1/getUser.feature [hunk 4]
```diff
@@ -122,19 +122,19 @@ Feature: get user
...
     Then the OCS status code should be "100"
     And the HTTP status code should be "200"
     And the display name returned by the API should be "New User"
     And the quota definition returned by the API should be "default"
 
   Scenario: a mixed-case normal user gets their own information, providing the mixed-case username in the URL
     Given these users have been created with default attributes and skeleton files:
-      | username | displayname |
-      | NewUser  | New User    |
-    When user "newuser" retrieves the information of user "NewUser" using the provisioning API
```
Comment (phil-davis) [inline tests/acceptance/features/apiProvisioning-v1/getUser.feature:137]: The mixed case has got lost here - see issue #38014

File: tests/acceptance/features/apiProvisioning-v2/getUser.feature [hunk 5]
```diff
@@ -92,19 +92,19 @@ Feature: get user
...
     Then the OCS status code should be "200"
     And the HTTP status code should be "200"
     And the display name returned by the API should be "New User"
     And the quota definition returned by the API should be "default"
 
   Scenario: a normal user gets their own information, providing uppercase username as authentication
     Given these users have been created with default attributes and skeleton files:
-      | username | displayname |
-      | newuser  | New User    |
-    When user "NEWUSER" retrieves the information of user "newuser" using the provisioning API
```
Comment (phil-davis) [inline tests/acceptance/features/apiProvisioning-v2/getUser.feature:107]: The uppercase has got lost here - see issue #38014

File: tests/acceptance/features/apiProvisioning-v2/getUser.feature [hunk 6]
```diff
@@ -113,9 +113,9 @@ Feature: get user
   @skipOnOcV10.3
   Scenario: a normal user gets their own information, providing uppercase username in the URL
     Given these users have been created with default attributes and skeleton files:
-      | username | displayname |
-      | newuser  | New User    |
-    When user "newuser" retrieves the information of user "NEWUSER" using the provisioning API
```
Comment (phil-davis) [inline tests/acceptance/features/apiProvisioning-v2/getUser.feature:118]: The uppercase has got lost here - see issue #38014

File: tests/acceptance/features/apiProvisioning-v2/getUser.feature [hunk 7]
```diff
@@ -124,19 +124,19 @@ Feature: get user
   @skipOnOcV10.3
   Scenario: a mixed-case normal user gets their own information, providing lowercase username in the URL
     Given these users have been created with default attributes and skeleton files:
-      | username | displayname |
-      | NewUser  | New User    |
-    When user "NewUser" retrieves the information of user "newuser" using the provisioning API
```
Comment (phil-davis) [inline tests/acceptance/features/apiProvisioning-v2/getUser.feature:129]: The mixed case has got lost here - see issue #38014

File: tests/acceptance/features/apiProvisioning-v2/getUser.feature [hunk 8]
```diff
@@ -124,19 +124,19 @@ Feature: get user
...
     Then the OCS status code should be "200"
     And the HTTP status code should be "200"
     And the display name returned by the API should be "New User"
     And the quota definition returned by the API should be "default"
 
   Scenario: a mixed-case normal user gets their own information, providing the mixed-case username in the URL
     Given these users have been created with default attributes and skeleton files:
-      | username | displayname |
-      | NewUser  | New User    |
-    When user "newuser" retrieves the information of user "NewUser" using the provisioning API
```
Comment (phil-davis) [inline tests/acceptance/features/apiProvisioning-v2/getUser.feature:139]: The mixed case has got lost here - see issue #38014

**Pull request merged**